src/process/path.h

```diff
@@ -11,8 +11,22 @@ typedef struct
     void *next_dentry;
     // the virtual fs mount where the path is mounted
     void *vfsmount;
+    // filter state machine state; init to 0 for filtering, -1 for none
+    int filter_state;
+    // filter match tag; filled by filter when a match is found
+    char filter_tag[MAX_TAG];
```

Comment:
maybe I am misunderstanding something but these tags don't need to be strings right? When userspace sets these tags it could just be a number that is meaningful to userspace. If userspace wants it as a string later on for tags they could have a separate conversion from "integer tags" to the "named tags". I don't think it matters all that much but it may make the kernel side be just a tad simpler (just a u16 or something) 🤷🏽 .

Reply:
Yeah, it would probably simplify things a bit.